onyo/commands/tests/test_new.py

```diff
@@ -86,6 +86,64 @@ def test_new_top_level(repo: OnyoRepo) -> None:
     fsck(repo)
 
 
+@pytest.mark.repo_dirs(*directories)
+def test_new_sub_dir_absolute_path(repo: OnyoRepo) -> None:
+    """
+    Test that `onyo new --path <path>` can create an asset in sub-directories of
+    the repository while the cwd is inside a different sub-directory.
+    """
+    # Change cwd to sub-directory
+    cwd = repo.git.root / 'spe\"cial\\char\'acteஞrs'
+    os.chdir(cwd)
+
+    # build a path for the new asset as an absolute path
+    file = repo.git.root / "just another path/laptop_apple_macbookpro.0"
+
+    ret = subprocess.run(['onyo', 'new', '--yes', '--path', file],
+                         capture_output=True, text=True)
+
+    # verify correct output
+    assert "The following will be created:" in ret.stdout
+    assert str(file) in ret.stdout
+    assert not ret.stderr
+    assert ret.returncode == 0
+
+    # verify that the new asset exists and the repository is in a clean state
+    assert len(repo.asset_paths) == 1
+    fsck(repo)
+
+
+@pytest.mark.repo_dirs(*directories)
+def test_new_sub_dir_relative_path(repo: OnyoRepo) -> None:
+    """
+    Test that `onyo new --path <path>` can create an asset in sub-directories of
+    the repository while the cwd is inside a different sub-directory with a
+    relative path.
+    """
+    # Change cwd to sub-directory
+    cwd = repo.git.root / 'r/e/c/u/r/s/i/v/e'
+    os.chdir(cwd)
+
+    # count levels until repo.git.root and build relative path with "../"
+    var = "../" * (len(cwd.relative_to(repo.git.root).parts) - 1)
+
+    # build a path for the new asset as a relative path
+    file = f"{var}just another path/laptop_apple_macbookpro.0"
```

Comment:
This building of a path seems a little weird. We use `pathlib`. Why sort of work around it, by manually constructing?

How about this instead?

```suggestion
    # build a path for the new asset as a relative path
    file = (repo.git.root / "just another path" / "laptop_apple_macbookpro.0").relative_to(cwd)
```

Reply:
I decided to do this way with strings, since this is a CLI test, and the CLI is called with a string.

Your example obviously makes sense too, because it is then converted into a string, during the subprocess call. But the code you share above will error out, since `relative_to(cwd)` wants one path to be a sub-path of the other (which it is not, therefor the error), and does not build a relative path with "../" in it.

Reply:
Sorry, I just posted the negative and forgot to post something productive, too!

I have now changed it to:
`file = Path(var, "just another path", "laptop_apple_macbookpro.0")`

This uses path, and keeps it relative with the `../../`. I assume you might dislike this still somewhat (because of the counting of levels in `var`), but I choose this approach because I think this works fairly well to test what the function should test.

If this is not satisfying, I am happy to discuss how this should actually look like :)